internal/corazarules/rule.go

```diff
@@ -76,3 +84,14 @@ func (r *RuleMetadata) Raw() string {
 func (r *RuleMetadata) SecMark() string {
 	return r.SecMark_
 }
+
+func (r *RuleMetadata) StrID() string {
+	if r.cachedStrID_ == "" {
```

Comment:
There aren't that many rules and I think this is used fairly commonly. Should we just populate it eagerly without the lazy pattern?

Reply:
Yeah it's a bigger change - I think it's fine to do but if it's too much for now, we can merge as is too

Reply:
In the end using lazy pattern leads to a race condition https://github.com/corazawaf/coraza/issues/1083